### PR TITLE
Enable nerfato to use thermoscene dataset

### PR DESCRIPTION
**Pull request by jiaruiyu99**

### CODE REVIEW

File: thermo_nerf/nerfacto_config/thermal_nerfacto_dataparser.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+        cy = []
+        height = []
+        width = []
+        distort = []
+
+        # sort the frames by fname
+        if self.config.use_thermal_scene:
+            key = "thermal_file_path"
+        else:
+            key = "file_path"
```
Comment (MalcolmMielle) [inline thermo_nerf/nerfacto_config/thermal_nerfacto_dataparser.py:85]: Are we making a whole new class because of just those four line?? I think it would be better to:

* Create a script where we convert a thermoScene transforms.json file to "classic" COLMAP file. We should load the "transforms.json" file and change the fey "file_path" for a specified key value before sending that update json file to nerfactor
* Send a patch upstream to nerfstudio to be able to specify yhe key for the path to be used.

I would propose doing the first option because it will help keep Nerfacto simpler and reading and rewritting a json is "cheap".

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

File: thermo_nerf/scripts/train_eval_script.py [hunk 2]
```diff
@@ -63,6 +67,18 @@ def __post_init__(self) -> None:
 if __name__ == "__main__":
     parameters = tyro.cli(TrainingParameters)
 
+    if parameters.model_type == ModelType.NERFACTO:
+        tmp_folder = Path("./data_set")
+        shutil.copytree(src=parameters.data, dst=tmp_folder)
+        os.chmod(tmp_folder / "transforms.json", stat.S_IRWXU)
```
Comment (MalcolmMielle) [inline thermo_nerf/scripts/train_eval_script.py:73]: Why is this needed?

File: thermo_nerf/scripts/train_eval_script.py [hunk 3]
```diff
@@ -63,6 +67,18 @@ def __post_init__(self) -> None:
 if __name__ == "__main__":
     parameters = tyro.cli(TrainingParameters)
 
+    if parameters.model_type == ModelType.NERFACTO:
+        tmp_folder = Path("./data_set")
```
Comment (MalcolmMielle) [inline thermo_nerf/scripts/train_eval_script.py:71]: Csan we name it `data_folder`?

File: thermo_nerf/scripts/train_eval_script.py (hunk 2 above)
Comment (jiaruiyu99) [reply thermo_nerf/scripts/train_eval_script.py:73]: The copytree command will copy the permission status, therefore without changing the permission, the json file is read only.

**Review by MalcolmMielle — APPROVED** [review]
